packages/flutter_driver/lib/src/driver/vmservice_driver.dart

```diff
@@ -0,0 +1,696 @@
+                 'calls enableFlutterDriverExtension() as the first call in main().',
+      );
+    } else if (isolate.pauseEvent is VMPauseExitEvent ||
+               isolate.pauseEvent is VMPauseBreakpointEvent ||
+               isolate.pauseEvent is VMPauseExceptionEvent ||
+               isolate.pauseEvent is VMPauseInterruptedEvent) {
+      // If the isolate is paused for any other reason, assume the extension is
+      // already there.
+      _log.trace('Isolate is paused mid-flight.');
+      await resumeLeniently();
+    } else if (isolate.pauseEvent is VMResumeEvent) {
+      _log.trace('Isolate is not paused. Assuming application is ready.');
+    } else {
+      _log.warning(
+        'Unknown pause event type ${isolate.pauseEvent.runtimeType}. '
+        'Assuming application is ready.'
+      );
+    }
+
+    // Invoked checkHealth and try to fix delays in the registration of Service
+    // extensions
+    Future<Health> checkHealth() async {
+      try {
+        // At this point the service extension must be installed. Verify it.
+        return await driver.checkHealth();
+      } on rpc.RpcException catch (e) {
+        if (e.code != error_code.METHOD_NOT_FOUND) {
+          rethrow;
+        }
+        _log.trace(
+          'Check Health failed, try to wait for the service extensions to be'
+          'registered.'
+        );
+        await enableIsolateStreams();
+        await waitForServiceExtension();
+        return driver.checkHealth();
+      }
+    }
+
+    final Health health = await checkHealth();
+    if (health.status != HealthStatus.ok) {
+      await client.close();
+      throw DriverError('Flutter application health check failed.');
+    }
+
+    _log.info('Connected to Flutter application.');
+    return driver;
+  }
+
+  static int _nextDriverId = 0;
+
+  static const String _flutterExtensionMethodName = 'ext.flutter.driver';
+  static const String _setVMTimelineFlagsMethodName = 'setVMTimelineFlags';
+  static const String _getVMTimelineMethodName = 'getVMTimeline';
+  static const String _clearVMTimelineMethodName = 'clearVMTimeline';
+  static const String _collectAllGarbageMethodName = '_collectAllGarbage';
+
+  // The additional blank line in the beginning is for _log.warning.
+  static const String _kDebugWarning = '''
+
+┏╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍┓
+┇ ⚠    THIS BENCHMARK IS BEING RUN IN DEBUG MODE     ⚠  ┇
+┡╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍┦
+│                                                       │
+│  Numbers obtained from a benchmark while asserts are  │
+│  enabled will not accurately reflect the performance  │
+│  that will be experienced by end users using release  ╎
+│  builds. Benchmarks should be run using this command  ┆
+│  line:  flutter drive --profile test_perf.dart        ┊
+│                                                       ┊
+└─────────────────────────────────────────────────╌┄┈  🐢
+''';
```

Comment:
this should presumably happen in every driver; it doesn't seems specific to the vmservice?